std/container/fixed_array.d

```diff
@@ -0,0 +1,504 @@
+
+    }
+
+
+/**
+    Constructor taking a random access range of type $(D Store).
+
+    Params:
+        store = $(XREF2 range, isRandomAccessRange, random access range) used as a store
+        initialLength = the initial length of the FixedArray. If set two $(D 0),
+                        all elements in store will be overwritten and if set to
+                        $(D range.length) no elements can be inserted into the
+                        $(D FixedArray) until some are removed.
+     */
+    this(Store store, size_t initialLength = size_t.max)
+    {
+
+        payload.store = move(store);
+        length = min(initialLength, store.length);
+    }
+
+
+/**
+Comparison for equality.
+
+    Two FixedArrays are equal, if they have the same length and
+    corresponding elements are equal.
+     */
+    bool opEquals(FixedArray rhs)
+    {
+        return opEquals(rhs);
+    }
+
+    /// ditto
+    bool opEquals(ref FixedArray rhs)
+    {
+        if(length != rhs.length)
+            return false;
+        return equal(store[0 .. length], rhs.store[0 .. length]);
+    }
+
+static if(is(typeof(store.dup)))
+{
+/**
+Duplicates the container. The elements themselves are not transitively
+duplicated. The underlying store must support $(D .dup) as well.
+
+Complexity: $(BIGOH n).
+     */
+    @property FixedArray dup()
+    {
+        return FixedArray(store.dup, length);
+    }
+}
+
+
+
+/**
+Property returning $(D true) if and only if the container has no
+elements.
+
+Complexity: $(BIGOH 1)
+     */
+    @property bool empty() const
+    {
+        return length == 0;
+    }
+
+/**
+Returns the number of elements in the container.
+
+Complexity: $(BIGOH 1).
+     */
+    @property size_t length() const
+    {
+        return payload.length;
+    }
+
+    /// ditto
+    alias opDollar = length;
+
+/**
+Set length.
+Params:
+    nlength = the new length
+Precondition: $(D nlength <= capacity)
+
+Complexity: $(BIGOH 1).
+     */
+
+    @property void length(size_t nlength) 
+    {
+        version(assert) if (nlength > capacity) throw new RangeError();
+        // TODO initialize the new elements?
+        payload.length = nlength;
+    }
+/**
+Returns the maximum number of elements the container can store 
+
+Complexity: $(BIGOH 1)
+     */
+    @property size_t capacity() const
+    {
+        return store.length;
+    }
+
+/**
+Returns a $(XREF range, isRandomAccessRange, random access range)
+of type $(LREF Range) that iterates over elements of the container, in
+forward order.
+
+Complexity: $(BIGOH 1)
+     */
+    Range opIndex()
+    {
+        return store[0 .. length];
+    }
+
+/**
+Returns a range that iterates over elements of the container from
+index $(D i) up to (excluding) index $(D j).
+
+Params:
+    i = index of first element in result
+    j = index of first element not in result.
+
+Precondition: $(D i <= j && j <= length)
+
+Complexity: $(BIGOH 1)
+     */
```

Comment:
So there's no empty line after this, which I submit as evidence for the request that it should be this way everywhere.

But the whitespace before `*/` seems random - please remove.